src/services/ansibleSyntaxCheck.ts

```diff
@@ -0,0 +1,197 @@
+import * as child_process from 'child_process';
+import { ExecException } from 'child_process';
+import { URL } from 'url';
+import { promisify } from 'util';
+import {
+  Connection,
+  Diagnostic,
+  DiagnosticSeverity,
+  DidChangeWatchedFilesParams,
+  Position,
+  Range,
+} from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { IAnsibleLintConfig } from '../interfaces/ansibleLintConfig';
+import { withInterpreter } from '../utils/misc';
+import { WorkspaceFolderContext } from './workspaceManager';
+const exec = promisify(child_process.exec);
+
+/**
+ * Acts as and interface to ansible syntax check and a cache of its output.
+ *
+ * ansible syntax-check may provide diagnostics for more than just the file for which
+ * linting was triggered, and this is reflected in the implementation.
+ */
+export class AnsibleSyntaxCheck {
+  private connection: Connection;
+  private context: WorkspaceFolderContext;
+  private useProgressTracker = false;
+
+  private configCache: Map<string, IAnsibleLintConfig> = new Map();
+
+  constructor(connection: Connection, context: WorkspaceFolderContext) {
+    this.connection = connection;
+    this.context = context;
+    this.useProgressTracker =
+      !!context.clientCapabilities.window?.workDoneProgress;
+  }
+
+  /**
+   * Perform ansible syntax-check for the given document.
+   *
+   * In case no errors are found for the current document, and linting has been
+   * performed on opening the document, then only the cache is cleared, and not
+   * the diagnostics on the client side. That way old diagnostics will persist
+   * until the file is changed. This allows inspecting more complex errors
+   * reported in other files.
```

Comment:
It doesn't apply here... actually it doesn't even apply to the origin (where it was copied from) - it's a leftover after a refactoring ;)
So now it's a leftover of a leftover :D